geode-core/src/main/java/org/apache/geode/cache/query/internal/ExecutionContext.java

```diff
@@ -127,13 +133,8 @@ public ExecutionContext(Object[] bindArguments, InternalCache cache) {
     this.bindArguments = bindArguments;
     this.cache = cache;
     this.cancelationTask = Optional.empty();
-
-    // Authorization & authentication logic happens on server side only.
-    if (cache.isClient()) {
-      this.methodInvocationAuthorizer = QueryConfigurationServiceImpl.getNoOpAuthorizer();
-    } else {
-      this.methodInvocationAuthorizer = cache.getQueryService().getMethodInvocationAuthorizer();
-    }
+    this.queryConfigurationService = cache.getService(QueryConfigurationService.class);
```

Comment:
Can this be null. What if user overrides the QueryConfigurationService provided with product.

Reply:
I don't think this service should be configurable/exchangeable by the user, meaning that "it's okay to fail" if the user mess around with it. I'm not completely sure, though, do you any extra thoughts about it?.

Reply:
We're not checking for the nullity of the `QueryConfigurationService` anywhere in the code, BTW, and maybe we should invest some time on investigating it... but I think that should belong to another PR and/or ticket, anyway, just my two cents 👍

Reply:
I think Anil has a good point, is there any chance the QueryConfigurationService is null (not set due to not being used?)  If no, then I think the null checks shouldn't be added.  If there is a chance for it to be null, maybe we should install a no-op service instead, so we can avoid null checks?

Reply:
@agingade @jhuynh1: done!

Reply:
The "MethodInvocationAuthorizer" returned by the "QueryConfigurationService" could be null as its implemented by users. This is may never happen; but we currently don't check for it.